src/node_webstorage.cc

```diff
@@ -0,0 +1,661 @@
+  r = sqlite3_step(stmt);
+  if (r == SQLITE_ROW) {
+    CHECK(sqlite3_column_type(stmt, 0) == SQLITE_BLOB);
+    value = String::NewFromUtf8(
+        env()->isolate(),
+        reinterpret_cast<const char*>(sqlite3_column_blob(stmt, 0)),
+        v8::NewStringType::kNormal,
+        static_cast<int>(sqlite3_column_bytes(stmt, 0))).ToLocalChecked();
+  } else if (r != SQLITE_DONE) {
+    THROW_SQLITE_ERROR(env(), r);
+  }
+
+  return value;
+}
+
+Local<Value> Storage::LoadKey(const int index) {
+  if (!Open()) {
+    return Local<Value>();
+  }
+
+  static const char sql[] =
+      "SELECT key FROM nodejs_webstorage LIMIT 1 OFFSET ?";
+  sqlite3_stmt* stmt = nullptr;
+  auto cleanup = OnScopeLeave([&]() { sqlite3_finalize(stmt); });
+  int r = sqlite3_prepare_v2(db_, sql, -1, &stmt, 0);
+  CHECK_ERROR_OR_THROW(env(), r, SQLITE_OK, Local<Value>());
+  r = sqlite3_bind_int(stmt, 1, index);
+  CHECK_ERROR_OR_THROW(env(), r, SQLITE_OK, Local<Value>());
+
+  auto value = Local<Value>();
+  r = sqlite3_step(stmt);
+  if (r == SQLITE_ROW) {
+    CHECK(sqlite3_column_type(stmt, 0) == SQLITE_BLOB);
+    value = String::NewFromUtf8(
+        env()->isolate(),
+        reinterpret_cast<const char*>(sqlite3_column_blob(stmt, 0)),
+        v8::NewStringType::kNormal,
+        static_cast<int>(sqlite3_column_bytes(stmt, 0))).ToLocalChecked();
+  } else if (r != SQLITE_DONE) {
+    THROW_SQLITE_ERROR(env(), r);
+  }
+
+  return value;
+}
+
+bool Storage::Remove(Local<Name> key) {
+  if (key->IsSymbol()) {
+    auto symbol_map = symbols_.Get(env()->isolate());
+    Maybe<bool> result = symbol_map->Delete(env()->context(), key);
+    return !result.IsNothing();
+  }
+
+  if (!Open()) {
+    return false;
+  }
+
+  static const char sql[] = "DELETE FROM nodejs_webstorage WHERE key = ?";
+  sqlite3_stmt* stmt = nullptr;
+  auto cleanup = OnScopeLeave([&]() { sqlite3_finalize(stmt); });
+  int r = sqlite3_prepare_v2(db_, sql, -1, &stmt, 0);
+  CHECK_ERROR_OR_THROW(env(), r, SQLITE_OK, false);
+  node::Utf8Value utf8key(env()->isolate(), key);
+  r = sqlite3_bind_blob(
+      stmt, 1, utf8key.out(), utf8key.length(), SQLITE_STATIC);
+  CHECK_ERROR_OR_THROW(env(), r, SQLITE_OK, false);
+  CHECK_ERROR_OR_THROW(env(), sqlite3_step(stmt), SQLITE_DONE, false);
+  return true;
+}
+
+bool Storage::Store(Local<Name> key, Local<Value> value) {
+  if (key->IsSymbol()) {
+    auto symbol_map = symbols_.Get(env()->isolate());
+    MaybeLocal<Map> result = symbol_map->Set(env()->context(), key, value);
+    return !result.IsEmpty();
+  }
+
+  Local<String> val;
+  if (!value->ToString(env()->context()).ToLocal(&val)) {
+    return false;
+  }
+
+  if (!Open()) {
+    return false;
+  }
+
+  static const char sql[] =
+      "INSERT INTO nodejs_webstorage (key, value, size) VALUES (?, ?, ?)"
+      "  ON CONFLICT (key) DO UPDATE"
+      "  SET value = EXCLUDED.value, size = EXCLUDED.size"
+      "  WHERE EXCLUDED.key = key";
+  sqlite3_stmt* stmt = nullptr;
```

Comment:
What takes ownership of the `stmt` here? Does this need to be cleared up? If nothing else, a comment here explaining would be helpful.